packages/linter/src/rule.ts

```diff
@@ -8,39 +8,52 @@ export interface RuleConstructor {
 }
 
 export abstract class Rule {
-  static href = "";
-  static info = "";
   abstract run(context: Context): Promise<Result | Array<Result>>;
-  protected async pass(s?: string) {
+  public meta() {
     return {
-      status: Status.PASS,
-      message: s,
-      url: Rule.href,
-      description: Rule.info
+      url: "",
+      title:
+        this.constructor.name.replace(
+          /([a-z])([A-Z])/g,
+          (_, c1, c2) => `${c1} ${c2.toLowerCase()}`
+        ) + "?",
+      info: ""
     };
   }
+  protected async pass(s?: string) {
```

Comment:
why are these methods `async`?

Reply:
It's because `run()` returns a promise--the test could be async, and often is. However, if it's not, and `pass()` was async, then you'd need to return `Promise.resolve(this.pass())` to satisfy the signature.

It does feel slightly weird though … I'm not 100% happy with the design of the linter.